framework/src/main/java/org/checkerframework/framework/type/StructuralEqualityVisitHistory.java

```diff
@@ -0,0 +1,57 @@
+package org.checkerframework.framework.type;
+
+import java.util.Optional;
+import javax.lang.model.element.AnnotationMirror;
+
+/** Stores the result of {@link StructuralEqualityComparer} for type arguments. */
```

Comment:
It would be useful to get some high-level information about how this differs from the `SubtypeVisitHistory` and what the true/false histories are for.